### PR TITLE
Feature/navbar login logout

### PR DESCRIPTION
**Pull request by No3ml3**
## Description

login logout on navbar

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [x] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [x] I tested admin by manually adding content from zero
- [x] I created easy to use admin experience which is self-explanatory
- [x] I added description to admin fields in hard-to-understand areas
- [x] I followed guidelines for naming `Typescript` variables
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by jaalazard — DISMISSED** [review]

**Review by Cardamced — DISMISSED** [review]
Top !

**Review by jaalazard — DISMISSED** [review]

**Review by Cardamced — DISMISSED** [review]

File: packages/frontend/web/src/components/Navbar.tsx [hunk 1]
```diff
@@ -119,6 +176,7 @@ export default function Navbar() {
               <Link to={content.url}>
                 <content.icon
                   className={`peer h-7 w-7 cursor-pointer sm:my-auto sm:me-3 text-[${color}] hover:text-[${hover}]`}
+                  {...(content.name === 'Logout' && { onClick: handleLogout })}
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Navbar.tsx]: Oh c'est nouveau ça
Syntaxe pas très lisible
Qu'est-ce qui se passe ? Je suis redirigé sur le lien ou la fonction s'execute ?

File: packages/frontend/web/src/components/Navbar.tsx [hunk 2]
```diff
@@ -51,10 +56,62 @@ const navbarContent: NavbarContent[] = [
...
+          name: 'Register',
+          url: '/register',
+          icon: UserPlus,
+        },
+        ...navbarContentInfo,
+      ]);
+    }
+  }, [isLoggedIn]);
+
+  console.log(navbarContent);
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Navbar.tsx]: console log

File: packages/backend/api/src/auth.ts [hunk 3]
```diff
@@ -56,18 +56,26 @@ authRouter.get('/check', async (req, res) => {
       issuer: 'http://localhost',
       audience: 'http://localhost',
     });
+
+    if (!check.payload || typeof check.payload.sub !== 'string') {
+      return res.json({
+        ok: false,
+        isLoggedIn: false,
+      });
+    }
```
Comment (YDrogen) [inline packages/backend/api/src/auth.ts:65]: T'es sûr de cette condition ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/components/Navbar.tsx (hunk 1 above)
Comment (No3ml3) [reply packages/frontend/web/src/components/Navbar.tsx]: les deux ... je devrais faire une page logout ? et redirigé vers la page logout ? je ne peux pas avoir un fetch et une redirection vers home ??

File: packages/backend/api/src/auth.ts (hunk 3 above)
Comment (No3ml3) [reply packages/backend/api/src/auth.ts:65]: oui pk ?

**Review by YDrogen — DISMISSED** [review]

File: packages/frontend/web/src/components/Navbar.tsx (hunk 1 above)
Comment (YDrogen) [reply packages/frontend/web/src/components/Navbar.tsx]: Non, pas de page logout, ça sert à rien
Si, mais pas en même temps; comment tu sais que la fonction se fait bien avant de rediriger ?

**Review by jaalazard — APPROVED** [review]

**Review by Justicesvr — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**